src/Middleware.php

```diff
@@ -3,31 +3,127 @@
 namespace Inertia;
 
 use Closure;
-use Illuminate\Support\Facades\Response;
-use Symfony\Component\HttpFoundation\RedirectResponse as Redirect;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 
-class Middleware
+abstract class Middleware
```

Comment:
Does it need to be abstract? Like technically if someone wanted to use it straight up, they could.

Reply:
Yeah, that's true. 

My thinking here was that it would provide a single way to do it, but maybe that's indeed a bit too limiting because we're not doing anything special here.